pkg/api/types.go

```diff
@@ -888,9 +888,8 @@ func (o *OrchestratorProfile) IsMetricsServerEnabled() bool {
 }
 
 // IsContainerMonitoringEnabled checks if the container monitoring addon is enabled
-func (o *OrchestratorProfile) IsContainerMonitoringEnabled() bool {
+func (k *KubernetesConfig) IsContainerMonitoringEnabled() bool {
```

Comment:
Moved this to be a method of KubernetesConfig (this is more idiomatic with other "is this addon enabled?" functionality) as a part of this work